lib/diagnostic.js

```diff
@@ -6,29 +6,55 @@
 // splits it into an array of normalized words
 // (uppercase strings without punctuation)
 // and returns that array.
-const normalizeWords = function (/* */) {
-  /* your response here */
+const normalizeWords = function (paragraph) {
+  let words = paragraph.split(/\s+/);
+  for (let i = 0; i < words.length; i++) {
+  let word = words[i].replace(/\W+/, '');
+  normalizeWords[i] = word.toUpperCase();
+}
 };
 
 // 2. Write a function that takes a string argument
 // and returns an array of unique normalized words.
-const uniqueWords = function (/* */) {
-  /* your response here */
+const uniqueWords = function (normalizedWords) {
+  let uniqueWordsAsKeys = {};
+for (let i = 0; i < normalizedWords.length; i++) {
+  let word = normalizedWords[i];
+  uniqueWordsAsKeys[word] = true;
+}
```

Comment:
I assume this is not finished? Please take a look at the lab from js-reference-types again: https://github.com/ga-wdi-boston/js-reference-types/blob/solution/lib/collections.js